Errors generating reference docs for v12 release

### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

`npm run docs` fails with `"Cannot find module 'ajv/dist/core'"`, which appears to be because of https://github.com/nrwl/nx/issues/27683#issuecomment-2870923541

Manually installing `ajv` directly worked: `npm i ajv`. However, we don't want to depend directly on that package long-term and it should not be necessary since it is an optional dependency.

One of the `ajv-validator` maintainers [said](https://github.com/ajv-validator/ajv/issues/2452#issuecomment-2169421394) that this was probably an npm issue, because it works with yarn and pnpm.

Once api-extractor runs, there are three warnings:
```
Warning: dist/core/block.d.ts:545:8 - (tsdoc-undefined-tag) The TSDoc tag "@yields" is not defined in this configuration
Warning: dist/core/field.d.ts:92:5 - (ae-setter-with-docs) The doc comment for the property "size_" must appear on the getter, not the setter.
Warning: dist/core/field_input.d.ts:69:5 - (ae-setter-with-docs) The doc comment for the property "size_" must appear on the getter, not the setter.
```

### Reproduction steps

1. 
2. 
3.


### Stack trace

```shell

```

### Screenshots

_No response_

### Browsers

_No response_

--- a/core/block.ts
+++ b/core/block.ts
@@ -1126,7 +1126,7 @@ export class Block {
   /**
    * Returns a generator that provides every field on the block.
    *
-   * @yields A generator that can be used to iterate the fields on the block.
+   * @returns A generator that can be used to iterate the fields on the block.
    */
   *getFields(): Generator<Field, undefined, void> {
     for (const input of this.inputList) {

--- a/core/field.ts
+++ b/core/field.ts
@@ -119,9 +119,6 @@ export abstract class Field<T = any>
     return this.size;
   }
 
-  /**
-   * Sets the size of this field.
-   */
   protected set size_(newValue: Size) {
     this.size = newValue;
   }

--- a/core/field_input.ts
+++ b/core/field_input.ts
@@ -102,11 +102,9 @@ export abstract class FieldInput<T extends InputTypes> extends Field<
    */
   override SERIALIZABLE = true;
 
-  /**
-   * Sets the size of this field. Although this appears to be a no-op, it must
-   * exist since the getter is overridden below.
-   */
   protected override set size_(newValue: Size) {
+    // Although this appears to be a no-op, it must exist since the getter is
+    // overridden below.
     super.size_ = newValue;
   }
 

--- a/eslint.config.mjs
+++ b/eslint.config.mjs
@@ -88,7 +88,8 @@ function buildTSOverride({files, tsconfig}) {
       '@typescript-eslint/no-explicit-any': ['off'],
       // We use this pattern extensively for block (e.g. controls_if) interfaces.
       '@typescript-eslint/no-empty-object-type': ['off'],
-
+      // TSDoc doesn't support @yields, so don't require that we use it.
+      'jsdoc/require-yields': ['off'],
       // params and returns docs are optional.
       'jsdoc/require-param-description': ['off'],
       'jsdoc/require-returns': ['off'],

